### PR TITLE
Disable rebalance on Gracefulshutdown and prevent Infinispan operations

### PR DESCRIPTION
**Pull request by ryanemerson**

### CODE REVIEW

File: pkg/reconcile/pipeline/infinispan/handler/manage/upgrades.go [hunk 1]
```diff
@@ -185,12 +179,17 @@ func GracefulShutdown(i *ispnv1.Infinispan, ctx pipeline.Context) {
...
 		ctx.Requeue(
 			ctx.UpdateInfinispan(func() {
+				i.Status.Replicas = &i.Spec.Replicas
 				i.Status.ReplicasWantedAtRestart = 0
 				i.SetCondition(ispnv1.ConditionGracefulShutdown, metav1.ConditionFalse, "")
+				// Add ConditionScalingUp so that on requeue we update the ClusterService so that no pods will be selected
+				i.SetCondition(ispnv1.ConditionScalingUp, metav1.ConditionTrue, "")
 			}),
 		)
+		return
```
Comment (rigazilla) [inline pkg/reconcile/pipeline/infinispan/handler/manage/upgrades.go:192]: I guess this is intentional

File: pkg/infinispan/client/v13/container.go [hunk 2]
```diff
@@ -82,74 +80,34 @@ func (c *container) Backups() api.Backups {
...
-	task = strings.ReplaceAll(task, "\n", "")
-	task = strings.ReplaceAll(task, "\t", "")
-
-	rsp, err := c.Post(url, task, headers)
-	defer func() {
-		err = httpClient.CloseBody(rsp, err)
-	}()
-	if err = httpClient.ValidateResponse(rsp, err, "Uploading GracefulShutdownTask", http.StatusOK); err != nil {
-		return
-	}
```
[the remark this answers is not in the document]
Comment (ryanemerson) [reply pkg/infinispan/client/v13/container.go:145]: I forgot to create an upstream issue with the context, sorry about that.

Please see https://github.com/infinispan/infinispan-operator/issues/1788 for the context. I'll also update the commit message to include a link to this issue.

File: pkg/reconcile/pipeline/infinispan/handler/manage/upgrades.go (hunk 1 above)
Comment (ryanemerson) [reply pkg/reconcile/pipeline/infinispan/handler/manage/upgrades.go:192]: That's a leftover from an earlier approach that I implemented, I'll remove.

File: pkg/reconcile/pipeline/infinispan/handler/manage/manage.go [hunk 3]
```diff
@@ -119,3 +119,83 @@ func ConfigureLoggers(infinispan *ispnv1.Infinispan, ctx pipeline.Context) {
...
+
+		if i.Spec.Replicas != 0 && *i.Status.Replicas != statefulSet.Status.CurrentReplicas {
+			// Remove all that have an index between CurrentReplicas and i.Spec.Replicas
+			pvcList := &corev1.PersistentVolumeClaimList{}
+			if err := ctx.Resources().List(i.PodSelectorLabels(), pvcList); err != nil {
+				ctx.Requeue(fmt.Errorf("unable to retrieve PVCs when scaling down: %w", err))
+				return
+			}
+
+			for idx := len(pvcList.Items) - 1; idx >= int(statefulSet.Status.CurrentReplicas); idx-- {
```
[the remark this answers is not in the document]
Comment (ryanemerson) [reply pkg/reconcile/pipeline/infinispan/handler/manage/manage.go]: Thinking about this more, there was no need to retrieve the actual PVCs list as the PVC names are deterministic. So I now just iterate over the range of PVC indexes that shouldn't exist and remove them, ignoring NotFound errors if encountered.

**Review by rigazilla — APPROVED** [review]
overall LGTM

**Pull request merged**